Dockerfile.vcsim

```diff
@@ -1,4 +1,9 @@
+FROM golang:latest as build
+WORKDIR /go/src/app
+RUN CGO_ENABLED=0 go get -u github.com/vmware/govmomi/vcsim
+
 FROM scratch
-LABEL maintainer="fabio@vmware.com"
-COPY vcsim /
-ENTRYPOINT [ "/vcsim" ]
\ No newline at end of file
+ENV TMPDIR=/tmp
+COPY --from=build /tmp /tmp
```

Comment:
Is this an empty dir? Btw: this might not work when running as non-root. It would be great to get non-root mode in this PR as well, if possible since several platforms these days enforce non-root execution.

Reply:
just saw your comment in the issue:

COPY --chown=appuser --from=build /tmp /tmp

can you please add a vcsim user/group and copy /etc/passwd as well and assume this role in scratch? 🙏

Reply:
I have revamped the Dockerfile.  Yes, this is an empty directory.

![image](https://user-images.githubusercontent.com/387006/110353485-1d516f00-8005-11eb-8254-59c3c87bf5e3.png)